lib/zone.ts

```diff
@@ -323,6 +329,8 @@ interface _ZonePrivate {
   patchEventTargetMethods:
       (obj: any, addFnName?: string, removeFnName?: string, metaCreator?: any) => boolean;
   patchOnProperties: (obj: any, properties: string[]) => void;
+  beforeRunTask: (zone: Zone, task: Task) => BeforeRunTaskStatus;
```

Comment:
I am not a fan of `beforeRunTask` and `afterRunTask`, but I do understand why you need them. But I think it would mean that we should also add them to `ZoneDelegate`? Or are you planning on making them private?

How do we communicate to the developer that this particular Task can't be wrapped? It never has `runTask` executed on it. It can never be in `fakeAsync`zone.